rtsp.go

```diff
@@ -255,14 +255,44 @@ func (rc *rtspCamera) closeConnection() {
 	}
 }
 
+// reconnectClientWithFallbackTransports attempts to setup the RTSP client with the given codec
+// using the transports in the order of TCP, UDP, and UDP Multicast. This overrides gortsplib's
+// default behavior of trying UDP first.
+func (rc *rtspCamera) reconnectClientWithFallbackTransports(codecInfo videoCodec) error {
+	// Define the transport preferences in order.
+	transportTCP := gortsplib.TransportTCP
+	transportUDP := gortsplib.TransportUDP
+	transportUDPMulticast := gortsplib.TransportUDPMulticast
+	transports := []*gortsplib.Transport{
+		&transportTCP,
+		&transportUDP,
+		&transportUDPMulticast,
+	}
+	// Try to reconnect with each transport in the order defined above.
+	// If all attempts fail, return the last error.
+	var lastErr error
+	for _, transport := range transports {
+		if err := rc.reconnectClient(codecInfo, transport); err != nil {
+			rc.logger.Warnf("cannot reconnect to rtsp server using transport %s, err: %s", transport.String(), err.Error())
+			lastErr = err
+			continue
```

Comment:
Should we add a small timeout here just in case we might be hitting the reconnect button too fast?